app/controllers/users.js

```diff
@@ -144,35 +157,72 @@ class UsersController extends BaseController {
   async edit(ws, data) {
     const {
       id: requestId,
-      user_edit: { login, current_password, new_password },
+      user_edit: { current_password, new_password },
     } = data;
 
-    const currentUser = await User.findOne({ login });
+    const userId = this.sessionRepository.getSessionUserId(ws);
+    const currentUser = await User.findOne({ _id: userId });
     if (!currentUser) {
       throw new Error(ERROR_STATUES.USER_LOGIN_OR_PASS.message, {
         cause: ERROR_STATUES.USER_LOGIN_OR_PASS,
       });
     }
 
-    if (!(await currentUser.isValidPassword(current_password))) {
+    if (
+      current_password &&
+      !(await currentUser.isValidPassword(current_password))
+    ) {
       throw new Error(ERROR_STATUES.INCORRECT_CURRENT_PASSWORD.message, {
         cause: ERROR_STATUES.INCORRECT_CURRENT_PASSWORD,
       });
     }
 
-    const updateUser = new User({ login, password: new_password });
-    await updateUser.encryptAndSetPassword();
-    await User.updateOne(
-      { login },
-      {
-        $set: {
-          password_salt: updateUser.params.password_salt,
-          encrypted_password: updateUser.params.encrypted_password,
-          updated_at: new Date(),
-        },
-      }
+    let updateParam = { updated_at: new Date() };
+
+    if (new_password) {
+      const updateUser = new User({ password: new_password });
+      await updateUser.encryptAndSetPassword();
+
+      updateParam["password_salt"] = updateUser.params.password_salt;
+      updateParam["encrypted_password"] = updateUser.params.encrypted_password;
+    }
+
+    delete data.user_edit["new_password"];
+    delete data.user_edit["current_password"];
+
+    for (const field in data.user_edit) {
+      updateParam[field] = data.user_edit[field];
+    }
+
+    const updatedUser = new User(
+      (
+        await User.findOneAndUpdate({ _id: userId }, { $set: updateParam })
+      )?.value
     );
-    const updatedUser = await User.findOne({ login });
+
+    const email = data.user_edit.email;
+    const phone = data.user_edit.phone;
+    if (email) {
+      await this.matchedRepository.matchedContact(currentUser.visibleParams(), {
```

Comment:
Instead of 2 similar lines

```
await this.matchedRepository.matchedContact(currentUser.visibleParams(), {
        removeRecord: 0,
        field: "email",
      });
      await this.matchedRepository.matchedContact(updatedUser.visibleParams(), {
        addRecord: 1,
        field: "email",
      });
```

can we somehow replace it with only one? e.g.

```
  await this.matchedRepository.matchedContact(currentUser.visibleParams(), {
        replaceRecord: 1,
        field: "email",
      });
      ```
      
      ?